packages/core/src/utilities/getImageSliceDataForVolumeViewport.ts

```diff
@@ -0,0 +1,50 @@
+import { IVolumeViewport } from '../types';
+import getSliceRange from './getSliceRange';
+import getTargetVolumeAndSpacingInNormalDir from './getTargetVolumeAndSpacingInNormalDir';
+
+/**
+ * It calculates the number of slices and the current slice index for a given
+ * Volume viewport
+ * @param viewport - volume viewport
+ * @returns An object with two properties: numberOfSlices and imageIndex.
+ */
+function getImageSliceDataForVolumeViewport(viewport: IVolumeViewport): {
+  numberOfSlices: number;
+  imageIndex: number;
+} {
+  const camera = viewport.getCamera();
+
+  const { spacingInNormalDirection, imageVolume } =
+    getTargetVolumeAndSpacingInNormalDir(viewport, camera);
+
+  if (!imageVolume) {
+    return;
+  }
+
+  const { viewPlaneNormal, focalPoint } = camera;
+  const actor = viewport.getActor(imageVolume.volumeId);
+
+  if (!actor) {
+    console.warn('No actor found for with actorUID of', imageVolume.volumeId);
+  }
+
+  const { volumeActor } = actor;
+  const sliceRange = getSliceRange(volumeActor, viewPlaneNormal, focalPoint);
+
+  const { min, max, current } = sliceRange;
+
+  // calculate number of steps from min to max with current normal spacing in direction
+  const numberOfSlices = Math.floor((max - min) / spacingInNormalDirection + 1);
+
+  // calculate the imageIndex based on min, max, current
+  const imageIndex = Math.round(
```

Comment:
Shouldn't this be floor rather than round? If you are within the spacing between where slice 0 and slice 1 is defined, you are viewing slice 0, right?

Reply:
round also allows you to report a value past the last index.

e.g. slices go from 0-1, 1-2, 2-3 (3 slices, ultra simplified units), if you round a camera viewing depth 2.75, you'll get index 3, when its clearly index 2 being viewed.